File: frontend/src/shared/types/api/task.ts
export interface TaskDto {
  task_id: string
  workspace_id: string
  meeting_id: string | null
  title: string
  assignee_member_id: string | null
  status: string
  progress: number | null
  blocker: string | null
  due_date: string | null
  // 계약 §4.7-6 이 PR #59 로 반영돼 항상 응답에 들어온다. 값은 대부분 null 이다
  start_date: string | null
  notion_page_id: string | null
  created_at: string
  updated_at: string
}
export interface TaskHistoryDto {
  history_id: string
  task_id: string
  changed_field: string
  old_value: string | null
  new_value: string | null
  change_source: string
  changed_by: string | null
  is_auto: boolean
  is_rolled_back: boolean
  rolled_back_at: string | null
  created_at: string
}

File: backend/app/schemas/task.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models import TaskStatus


class TaskCreateRequest(BaseModel):
    workspace_id: str = Field(..., description="소속 워크스페이스 ID")
    title: str = Field(..., min_length=1, max_length=300)
    meeting_id: str | None = None
    assignee_member_id: str | None = None
    start_date: date | None = None
    due_date: date | None = None
    status: TaskStatus = TaskStatus.TODO
    progress: int | None = Field(None, ge=0, le=100)
    blocker: str | None = None
    created_by: str | None = Field(None, description="수동 생성한 PM member_id")


class TaskUpdateRequest(BaseModel):
    title: str | None = Field(None, min_length=1, max_length=300)
    assignee_member_id: str | None = None
    status: TaskStatus | None = None
    progress: int | None = Field(None, ge=0, le=100)
    blocker: str | None = None
    start_date: date | None = None
    due_date: date | None = None
    changed_by: str | None = Field(None, description="변경한 PM member_id")


class TaskResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    task_id: str
    workspace_id: str
    meeting_id: str | None
    title: str
    assignee_member_id: str | None
    status: str
    progress: int | None
    blocker: str | None
    start_date: date | None
    due_date: date | None
    notion_page_id: str | None
    created_at: datetime
    updated_at: datetime


class TaskListResponse(BaseModel):
    items: list[TaskResponse]
    total: int


class TaskHistoryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    history_id: str
    task_id: str
    changed_field: str
    old_value: str | None
    new_value: str | None
    change_source: str
    changed_by: str | None
    is_auto: bool
    is_rolled_back: bool
    rolled_back_at: datetime | None
    created_at: datetime


class TaskHistoryListResponse(BaseModel):
    items: list[TaskHistoryResponse]
    total: int
